### ISSUE TITLE
YT links with channel IDs not working

### ISSUE THREAD

**Issue by mundry** [whitep4nth3r/womenwhostream.tech#14]
Hey,

the YouTube links using a channel ID are not working. The `/c/` path is for custom URLs only. Navigating to the path with a channel ID results in a 404, despite the channel ID itself being of a valid channel. Channel IDs only work with the `/channel/` path. However, custom URLs don't. [1]

So there needs to be some sort of differentiation. I can't really think of a reliable way to determine programmatically whether a `youtubeChannelId` value is a custom URL or an ID, just from the value entered for `youtubeChannelId`. Either always use a channel ID (as not every channel has a custom URL or is eligible for one by default) or add a flag in Contentful denoting a channel ID as a custom URL, or even add a separate field in Contentful for YouTube custom URLs and channel IDs and fill in either or.

Perhaps there are other ways to handle this I can't think of, or even an URL format working with both custom URLs and channel IDs.

[1] https://support.google.com/youtube/answer/6180214?hl=en

**Comment by whitep4nth3r** [comment]
Thanks for reporting this @mundry!

I think a quick workaround would be to store the full URL to the Youtube channel in Contentful.

**Comment by OliverFlint** [comment]
> Thanks for reporting this @mundry!
> 
> I think a quick workaround would be to store the full URL to the Youtube channel in Contentful.

Agreed. We can enter `/channel/UCUZHFZ9jIKrLroW8LcyJEQQ` or `/c/YouTubeCreators`. I'd say that would be acceptable as a solution with some regex input validation in contentful too.
